common/functions/src/scalars/conditionals/in_basic.rs

```diff
@@ -127,26 +127,23 @@ impl<const NEGATED: bool> Function for InFunction<NEGATED> {
 
         let types: Vec<DataTypePtr> = columns.iter().map(|col| col.column().data_type()).collect();
 
-        let least_super_dt = if columns[0]
-            .field()
-            .data_type()
-            .data_type_id()
-            .is_date_or_date_time()
-        {
+        let data_type = remove_nullable(columns[0].field().data_type());
+        let least_super_dt = if data_type.data_type_id().is_date_or_date_time() {
             match columns[1..]
                 .iter()
                 .map(|column| column.field().data_type().data_type_id())
-                .all(|t| t.is_string() || t.is_date_or_date_time())
+                .all(|t| t.is_string() || t.is_date_or_date_time() || t.is_null())
             {
-                true => columns[0].field().data_type().clone(),
+                true => wrap_nullable(columns[0].field().data_type()),
```

Comment:
Why it need to be wrapped nullable?

Reply:
Column datatype may be nullable and the values in (*, *) may be NULL. In this case, the datatype should be wrapped nullable. But `wrap_nullable` is compatible without nullable, so i use wrap_nullable. For performance, it should add the judgement to choose. I will fix it to add nullable judgement.